Initial data load...need to check second quantitative variable

bessjc12 opened the pull request:
?? have annual contract revenue and congressional district as numerical variables.  not sure what final rendering of our project will look like with these two variables....I am wondering if percent of total contract revenue would be a related variable that would show well in scatter plot?

bessjc12 committed:
Initial data load...need to check second quantitative variable

cmgiven commented:
So I wouldn't say that congressional district is a quantitative variable. Even though it is a number, it's a categorical variable, and so we shouldn't use it on a continuous scale like a scatterplot.

I'm getting a bunch of errors in your code. All pretty easy to fix if you look in the console. Things like using `=` when you meant to use `+`. Did you mean to open a pull request yet?

cmgiven commented:
Also, note that we're expecting you to modify `index.html`, not create new HTML files. This works great because we're using Git, so we can easily see what's changed between versions.

bessjc12 commented:
Thanks Chris,  I did want to insinuate my code was ready.  I have work to do. I'll fix the name convention. 
Jonathan

Jonathan Bess
Sent from my iPhone
bessjc12@gmail.com

> On Jun 29, 2016, at 17:10, Chris Given notifications@github.com wrote:
> 
> So I wouldn't say that congressional district is a quantitative variable. Even though it is a number, it's a categorical variable, and so we shouldn't use it on a continuous scale like a scatterplot.
> 
> I'm getting a bunch of errors in your code. All pretty easy to fix if you look in the console. Things like using = when you meant to use +. Did you mean to open a pull request yet?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.

bessjc12 commented:
I'm going to create another quant variable. I see have other issues to update too   

Jonathan Bess
Sent from my iPhone
bessjc12@gmail.com

> On Jun 29, 2016, at 17:10, Chris Given notifications@github.com wrote:
> 
> So I wouldn't say that congressional district is a quantitative variable. Even though it is a number, it's a categorical variable, and so we shouldn't use it on a continuous scale like a scatterplot.
> 
> I'm getting a bunch of errors in your code. All pretty easy to fix if you look in the console. Things like using = when you meant to use +. Did you mean to open a pull request yet?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.

bessjc12 committed:
Week 4 Axis progress

bessjc12 committed:
original axis code

bessjc12 committed:
axis but no values?

bessjc12 commented:
axis rendering but no values.
following errors I can't seem to resolve:
d3.min.js:1 Error: <g> attribute transform: Trailing garbage, "translate(0,NaN)".i @ 
d3.min.js:1 Error: <text> attribute transform: Trailing garbage, "translate(660 ,)".i @ 
d3.min.js:1 Error: <g> attribute transform: Expected number, "translate([object Object]0…".i @ 
d3.min.js:1 Error: <text> attribute transform: Expected number, "translate(,360".i @ 
(index):241 Uncaught TypeError: Cannot read property 'selectAll' of undefined

bessjc12 commented:
Can't figure out why x axis is at top of chart vice bottom?

bessjc12 committed:
Axis rendered but selectAll points undefined?

bessjc12 commented:
So I fixed the x axis and removed transforms translates to work to get data points on chart.  Following error and I don't know where to look:
Uncaught TypeError: Cannot read property 'selectAll' of undefined
update @ (index):245(anonymous function) @ (index):144(anonymous function) @ d3.min.js:1t @ d3.min.js:1i @ d3.min.js:1

cmgiven commented on index.html:
```diff
@@ -89,46 +121,142 @@
 
-    svg = d3.select('#chart')
-      .append('svg');
+  var svg = d3.select('#chart')
```
This is the problem. Hopefully it's becoming clearer after this week's class demo, but `svg` is a global variable. We define it outside of `setup()` on line 127 so that it's available anywhere in our script. But when you use the `var` keyword here, you create a new _local_ variable with the same name, and this variable expires once `setup()` is finished executing. This is why why you try to use it inside `update()` you get the error about it being undefined: the global `svg` variable was never given a value, because this local `svg` swallowed it.

bessjc12 committed:
Axis and labels added.

bessjc12 committed:
revised data set second variable to indicate % contribution to overall spend

bessjc12 committed:
Background color

bessjc12 committed:
Added grid and changed button to call for state vice year

bessjc12 committed:
added percent to x axis values and grid lines

bessjc12 commented:
Getting caught up...Revised data to get more meaningful scatter gram.  Added grid lines and chart color.  Revised axis labels.  Changed filter to state but can not get the filter by state to work?

cmgiven commented:
Two issues here:
- Check line 293. Are you using the variable with the filtered data here?
- Your filter function is checking whether `options.State` is exactly the same as the `State` variable in the data. Check both sets of values, do they ever match?

bessjc12 committed:
histograms and filters functioning for x axis only

bessjc12 commented:
started revised code and now have x axis histograms functioning along with state filter function working.   Working on rest of presentation.

bessjc12 committed:
week 6 changes made to add two chart histograms with mouseover and filter functions on state

bessjc12 committed:
Graph with x and y histograms, filters and tooltips working.  Placement of y histogram off
